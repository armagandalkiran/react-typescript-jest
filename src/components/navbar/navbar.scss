<<<<<<< HEAD
@import "../../common.scss";

.navbarContainer{
    padding:20px;
    text-align: center;
    background-color: $primary;
}
=======
.navbarContainer {
  padding: 20px;
  text-align: center;
}
>>>>>>> 4705ecb (todoapp-task-1 navbar tests added)
