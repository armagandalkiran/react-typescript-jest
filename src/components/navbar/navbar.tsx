import { navbarItems } from "../../models/constants";
import "./navbar.scss";

export const Navbar = () => {
  return (
    <nav className="navbarContainer">
      <h1 className="navbarTitle">{navbarItems.title}</h1>
      <p className="navbarDescription">{navbarItems.description}</p>
    </nav>
  );
};
