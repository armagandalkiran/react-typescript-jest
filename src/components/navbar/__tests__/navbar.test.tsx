import { render, screen, cleanup } from "@testing-library/react";
import { Navbar } from "../navbar";

afterEach(() => {
  cleanup();
});

test("should render navbar component", () => {
  render(<Navbar />);
  const navbarElement = screen.getByTestId("navbar")
  const navbarTitleElement = screen.getByTestId("navbarTitle");
  const navbarDescriptionElement = screen.getByTestId("navbarDescription")
  expect(navbarElement).toBeInTheDocument();
  expect(navbarTitleElement ).toHaveTextContent("My ToDo list");
  expect(navbarDescriptionElement ).toHaveTextContent("lorem ipsum lorem ipsum lorem ipsum");
});
