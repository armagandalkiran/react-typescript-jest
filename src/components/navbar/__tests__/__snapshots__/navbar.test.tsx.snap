// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbarContainer"
  data-testid="navbar"
>
  <h1
    data-testid="navbarTitle"
  >
    My ToDo list
  </h1>
  <p
    data-testid="navbarDescription"
  >
    lorem ipsum lorem ipsum lorem ipsum
  </p>
</nav>
`;
